src/macterm.c

```diff
@@ -2379,10 +2420,14 @@ mac_draw_glyph_string (struct glyph_string *s)
 		  mac_set_foreground (s->gc, xgcv.foreground);
 		}
 	    }
-	  else if (s->face->underline == FACE_UNDERLINE_SINGLE)
+	  else if (s->face->underline == FACE_UNDERLINE_SINGLE
```

Comment:
ℹ️ I reused this branch for the thickness and position calculations, which I think we want for these styles as we do for a plain underline.